lib/quick_settings_dialog.dart

```diff
@@ -41,8 +115,80 @@ Widget quickSettingsDialog(BuildContext context) => Consumer<SettingsProvider>(
                   }),
                 ),
               ),
+              const SizedBox(height: 10),
             ],
           ),
         );
       },
     );
+
+class RelatedTile extends StatelessWidget {
+  RelatedTile({
+    required this.songLink,
+    required this.relatedReason,
+    required this.provider,
+    Key? key,
+  }) : super(key: key);
+
+  String songLink;
+  String relatedReason;
+  SettingsProvider provider;
+
+  @override
+  Widget build(BuildContext context) {
+    String songId = songLink.split("/").last;
+    Book book = songLink.split("/").first == '21' ? Book.blue : Book.black;
+
+    return ListTile(
+      leading: Card(
+          child: Padding(
+        padding: const EdgeInsets.all(7),
+        child: Text(
+          songId,
+          style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 17),
+        ),
+      )),
+      title: Text(relatedReason),
+      onTap: () async {
+        // HACK - needs refactor
```

Comment:
Hmm?

Reply:
Jelenleg jobb híján: a dialog kiszámolja, az ének hanyadik a mapben, ezt átadja, majd a songsPage visszaszámolja az indexből az eredeti keyt.
A főoldalra tekintettel működik így a songsPage, de nem így kell majd csinálni, ha lesznek listák.
Csak jeleztem hogy a refaktorba ezt a részt is bele kell majd venni, amikor megtörténik.